src/asset-swapper/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -1602,14 +1606,28 @@ export class SamplerOperations {
                                     ERC20BridgeSource.Balancer,
                                 ),
                             );
+                    case ERC20BridgeSource.Beethovenx:
                     case ERC20BridgeSource.BalancerV2: {
+                        if (
+                            this.chainId === ChainId.Optimism &&
+                            source === ERC20BridgeSource.Beethovenx &&
+                            (takerToken === OPTIMISM_TOKENS.OP || makerToken === OPTIMISM_TOKENS.OP)
+                        ) {
+                            // BeethovenX on Optimism is currently seeing some revert issues with the OP token
+                            // so we're going to blacklist the token until we can investigate.
+                            return [];
+                        }
+
                         const cache = this.poolsCaches[source];
                         if (!cache) {
                             return [];
                         }
 
                         const swaps = cache.getCachedSwapInfoForPair(takerToken, makerToken);
-                        const vault = BALANCER_V2_VAULT_ADDRESS_BY_CHAIN[this.chainId];
+                        const vault =
+                            source === ERC20BridgeSource.BalancerV2
+                                ? BALANCER_V2_VAULT_ADDRESS_BY_CHAIN[this.chainId]
```

Comment:
nit/optional: maybe we can merge `BALANCER_V2_VAULT_ADDRESS_BY_CHAIN` and `BEETHOVEN_X_VAULT_ADDRESS_BY_CHAIN`